src/main/java/uk/gov/companieshouse/insolvency/delta/serialization/ChsDeltaSerializer.java

```diff
@@ -1,28 +1,57 @@
 package uk.gov.companieshouse.insolvency.delta.serialization;
 
+import java.nio.charset.StandardCharsets;
+
 import org.apache.avro.io.DatumWriter;
 import org.apache.avro.io.EncoderFactory;
 import org.apache.avro.specific.SpecificDatumWriter;
 import org.apache.kafka.common.serialization.Serializer;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 import uk.gov.companieshouse.delta.ChsDelta;
+import uk.gov.companieshouse.insolvency.delta.exception.NonRetryableErrorException;
 import uk.gov.companieshouse.kafka.exceptions.SerializationException;
 import uk.gov.companieshouse.kafka.serialization.AvroSerializer;
+import uk.gov.companieshouse.logging.Logger;
 
 @Component
-public class ChsDeltaSerializer implements Serializer<ChsDelta> {
+public class ChsDeltaSerializer implements Serializer<Object> {
+
+    private final Logger logger;
+
+    @Autowired
+    public ChsDeltaSerializer(Logger logger) {
+        this.logger = logger;
+    }
 
     @Override
-    public byte[] serialize(String var1, ChsDelta chsDelta) {
+    public byte[] serialize(String topic, Object payload) {
+        logger.trace("Payload serialised: " + payload);
+
         try {
-            DatumWriter<ChsDelta> writer = new SpecificDatumWriter<>();
-            EncoderFactory encoderFactory = EncoderFactory.get();
+            if (payload == null) {
+                return null;
+            }
+
+            if (payload instanceof byte[]) {
+                return (byte[]) payload;
```

Comment:
Probably better to write tests for different conditional blocks on serializer and de-serialzer